src/components/VersionSelector/VersionSelector.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import {
+  ArrowIcon,
+  StyledWrapper,
+  StyledButtonWrapper,
+  StyledButton,
+  StyledLabel,
+  StyledDescription,
+  StyledMenuList,
+  StyledDisplay,
+  StyledDropdown,
+} from './styled.elements';
+import { Option } from './Option';
+import { VersionSelectorProps } from './types';
+import { useOutsideClick } from './use-outside-click';
+
+/**
+ * Version Selector Dropdown component based structurally and stylistically off LG Select
+ */
+const VersionSelectorComponent = ({
+  resource_versions,
+  active,
+  root_url,
+  variant,
+  description,
+}: VersionSelectorProps): JSX.Element => {
+  const initialSelectedIdx = resource_versions.indexOf(active.resource_version);
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [selectedIdx, setSelectedIdx] = React.useState<number>(initialSelectedIdx);
+  const menuListRef = React.useRef(null);
+  useOutsideClick(menuListRef, () => {
+    if (open) setOpen(false);
+  });
+
+  const handleClick = (idx: number) => {
+    setSelectedIdx(idx);
+    setOpen(false);
+  };
+
+  React.useEffect(() => {
+    console.log(root_url + '/' + resource_versions[selectedIdx]);
+  }, [selectedIdx, resource_versions, root_url]);
+
+  return (
+    <StyledWrapper ref={menuListRef}>
+      <StyledButtonWrapper>
+        <StyledLabel>Version Selector: v{active.api_version}</StyledLabel>
+        {description && <StyledDescription>{description}</StyledDescription>}
+        <StyledButton onClick={() => setOpen(!open)}>
+          <StyledDisplay>
+            <div>
+              <div>
+                <span></span>
```

Comment:
What's the purpose of this empty span?

Reply:
Hmm, this sometimes has an icon here to the left of the selected value. I guess it has no real structural or stylistic purpose here and probably never will. Good catch!